src/App.tsx

```diff
@@ -2,16 +2,42 @@ import { useState } from "react";
 import { Route, RouterProvider, createBrowserRouter, createRoutesFromElements } from "react-router-dom";
 import "./App.css";
 import SplashScreen from "./components/SplashScreen";
+import FormTransition from "./pages/FormTransition";
 import Layout from "./pages/Layout";
-import PaniniCreatorLayout from "./pages/PaniniCreatorLayout";
+import PaniniCreator from "./pages/PaniniCreator";
+
+// 0 = start, restarted state, panini, restart
+export type UserStep = 0 | 1 | 2 | 3;
+export type SetOrderData = (val: object) => void;
+export type UpdateUserStep = () => void;
+
 function App() {
-  const [orderData, setOrderData] = useState({});
+  const [orderData, setOrderData] = useState<object>({});
+  const [userStep, setUserStep] = useState<UserStep>(0);
+  console.log(userStep);
+  const handleUserStepUpdate: UpdateUserStep = () => {
+    setUserStep((prev) => {
+      if (prev === 0) {
+        return 2;
+      } else if (prev < 3) {
+        return Math.round(prev + 1) as UserStep;
+      } else {
+        return 1;
+      }
+    });
+  };
+  const handleSetOrderData: SetOrderData = (val: object) => {
+    setOrderData(val);
+  };
   const router = createBrowserRouter(
     createRoutesFromElements(
-      <Route path="/" element={<Layout isPaniniOrdered={Object.keys(orderData).length > 0 ? true : false} />}>
-        <Route index element={<SplashScreen shouldTransition={false} />} />
-        <Route path="/panini_creator" element={<PaniniCreatorLayout shouldTransition={true} />}></Route>
-        <Route path="/panini_creator:options" element={<PaniniCreatorLayout shouldTransition={false} />}></Route>
+      <Route
+        path="/"
```

Comment:
**nitpick:** Your routes got very complicated - you should have 3 routes maximum here. Splash screen, form screen, and success screen.